src/services/rfqm_service.ts

```diff
@@ -553,12 +578,16 @@ export class RfqmService {
             return;
         }
 
-        // TODO: Submit To Chain
+        // submit to chain
+        await this.completeSubmissionLifecycleAsync(orderHash, workerAddress, calldata!);
 
-        // Update Status
+        // Update Status base on transaction submission status
+        const finalJobStatus = await this._getJobStatusFromSubmissionsAsync(orderHash);
```

Comment:
instead of adding this function (which adds another DB read), could we simply have `completeSubmissionLifecycleAsync` return the mined `RfqmTransactionSubmissionEntity`?

```
public async completeSubmissionLifecycleAsync(
        orderHash: string,
        workerAddress: string,
        callData: string,
    ): Promise<RfqmTransactionSubmissionEntity> {
    ....
   }
```

(or heck, even return all of them `Promise<RfqmTransactionSubmissionEntity[]>` )

Reply:
agreed